controllers/nginx/pkg/cmd/controller/metrics.go

```diff
@@ -0,0 +1,235 @@
+		"nginx_write_bytes_total",
+		"number of bytes written",
+		nil, nil)
+
+	memResidentbytesDesc = prometheus.NewDesc(
+		"nginx_resident_memory_bytes",
+		"number of bytes of memory in use",
+		nil, nil)
+
+	memVirtualbytesDesc = prometheus.NewDesc(
+		"nginx_virtual_memory_bytes",
+		"number of bytes of memory in use",
+		nil, nil)
+
+	startTimeDesc = prometheus.NewDesc(
+		"nginx_oldest_start_time_seconds",
+		"start time in seconds since 1970/01/01",
+		nil, nil)
+
+	activeDesc = prometheus.NewDesc(
+		"nginx_active_connections",
+		"total number of active connections",
+		nil, nil)
+
+	acceptedDesc = prometheus.NewDesc(
+		"nginx_accepted_connections",
+		"total number of accepted client connections",
+		nil, nil)
+
+	handledDesc = prometheus.NewDesc(
+		"nginx_handled_connections",
+		"total number of handled connections",
+		nil, nil)
+
+	requestsDesc = prometheus.NewDesc(
+		"nginx_total_requests",
+		"total number of client requests",
+		nil, nil)
+
+	readingDesc = prometheus.NewDesc(
+		"nginx_current_reading_connections",
+		"current number of connections where nginx is reading the request header",
+		nil, nil)
+
+	writingDesc = prometheus.NewDesc(
+		"nginx_current_writing_connections",
+		"current number of connections where nginx is writing the response back to the client",
+		nil, nil)
+
+	waitingDesc = prometheus.NewDesc(
+		"nginx_current_waiting_connections",
+		"current number of idle client connections waiting for a request",
+		nil, nil)
+)
+
+type (
+	scrapeRequest struct {
+		results chan<- prometheus.Metric
+		done    chan struct{}
+	}
+
+	namedProcessCollector struct {
+		scrapeChan chan scrapeRequest
+		*proc.Grouper
+		fs *proc.FS
+	}
+)
+
+func newProcessCollector(
+	children bool,
+	n common.MatchNamer) (*namedProcessCollector, error) {
+
+	fs, err := proc.NewFS("/proc")
+	if err != nil {
+		return nil, err
+	}
+	p := &namedProcessCollector{
+		scrapeChan: make(chan scrapeRequest),
+		Grouper:    proc.NewGrouper(children, n),
+		fs:         fs,
+	}
+	_, err = p.Update(p.fs.AllProcs())
+	if err != nil {
+		return nil, err
+	}
+
+	go p.start()
+
+	return p, nil
+}
+
+// Describe implements prometheus.Collector.
+func (p *namedProcessCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- cpuSecsDesc
+	ch <- numprocsDesc
+	ch <- readBytesDesc
+	ch <- writeBytesDesc
+	ch <- memResidentbytesDesc
+	ch <- memVirtualbytesDesc
+	ch <- startTimeDesc
+}
+
+// Collect implements prometheus.Collector.
+func (p *namedProcessCollector) Collect(ch chan<- prometheus.Metric) {
+	req := scrapeRequest{results: ch, done: make(chan struct{})}
+	p.scrapeChan <- req
+	<-req.done
+}
+
+func (p *namedProcessCollector) start() {
+	for req := range p.scrapeChan {
+		ch := req.results
+		p.scrape(ch)
+		req.done <- struct{}{}
+	}
+}
+
+func (p *namedProcessCollector) scrape(ch chan<- prometheus.Metric) {
+	s, err := getNginxStatus()
+	if err != nil {
+		glog.Warningf("unexpected error obtaining nginx status info: %v", err)
+		return
+	}
+
```

Comment:
this should be isolated to this backend. 
In case of caddy or haproxy there's already an exporter:
https://github.com/miekg/caddy-prometheus
https://github.com/prometheus/haproxy_exporter